src/change_stream/mod.rs

```diff
@@ -0,0 +1,172 @@
+//! Contains the functionality for ChangeStreams.
+pub mod document;
+pub mod option;
+
+use std::{marker::PhantomData, pin::Pin, task::Poll};
+
+use bson::Document;
+use futures::{stream::Stream, task::Context};
+use serde::{de::DeserializeOwned, Deserialize};
+
+use crate::{
+    change_stream::{
+        document::{ChangeStreamEventDocument, ResumeToken},
+        option::ChangeStreamOptions,
+    },
+    error::Result,
+    options::AggregateOptions,
+    selection_criteria::ReadPreference,
+    Client,
+    Collection,
+    Cursor,
+    Database,
+};
+
+/// A `ChangeStream` streams the ongoing changes of its associated collection, database or
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target.
+/// `ChangeStream`'s are "resumable", meaning that they can be restarted at a certain place in the
```

Comment:
can we add a blank line between this paragraph and the preceding one? Also, after re-reading this, I think it would be slightly clearer if we changed "certain" to "given".

Reply:
fixed